compiler/rustc_hir_analysis/src/hir_ty_lowering/bounds.rs

```diff
@@ -23,44 +23,206 @@ use crate::hir_ty_lowering::{
     AssocItemQSelf, FeedConstTy, HirTyLowerer, PredicateFilter, RegionInferReason,
 };
 
-impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
-    /// Add a `Sized` bound to the `bounds` if appropriate.
-    ///
-    /// Doesn't add the bound if the HIR bounds contain any of `Sized`, `?Sized` or `!Sized`.
-    pub(crate) fn add_sized_bound(
+impl<'tcx, 'a> dyn HirTyLowerer<'tcx> + '_
+where
+    'tcx: 'a,
```

Comment:
why here instead of in the method, why is it even required instead of just using `hir_bounds: &[hir::GenericBound<'tcx>]`?